src/lib.rs

```diff
@@ -3390,6 +3390,10 @@ impl Connection {
     /// On success the number of bytes written is returned, or [`Done`] if no
     /// data was written (e.g. because the stream has no capacity).
     ///
+    /// This methods also allows to send 0-length FIN `STREAM_FRAME` frames,
+    /// which means it is possible to use it with 0-length bytes while the
+    /// FIN flag is set. In this case it will return [`Ok(0)`].
```

Comment:
Maybe something like

```suggestion
    /// Applications can provide a 0-length buffer with the fin flag set to 
    /// true. This will lead to a 0-length FIN STREAM frame being sent at the 
    /// latest offset. This is the only case where [`Ok(0)`] is returned.
```